ev3dev.go

```diff
@@ -208,14 +208,75 @@ type StaterDevice interface {
 	State() (MotorState, error)
 }
 
+// TODO(kortschak) Remove this when pre-poll kernels are not supported.
+func init() {
+	b, err := ioutil.ReadFile("/proc/version")
+	if err != nil {
+		return
+	}
+	pollIsSupported = kernelSatisfies(minKernelVersion, string(chomp(b)))
+}
+
+var (
+	minKernelVersion = [4]int{4, 4, 19, 15}
+	pollIsSupported  bool
+)
+
+func kernelSatisfies(min [4]int, kernel string) bool {
+	const prefix = "Linux version "
+	if !strings.HasPrefix(kernel, prefix) {
+		return false
+	}
+	kernel = kernel[len(prefix):]
+	version := strings.SplitN(kernel, " ", 2)[0]
+	if !strings.Contains(version, "ev3dev") {
+		return false
+	}
+
+	// TODO(kortschak) Remove this when poll support PR is merged in ev3dev.
+	if !strings.Contains(version, "kortschak") {
+		return false
+	}
+
+	parts := strings.FieldsFunc(version, func(r rune) bool { return r == '.' || r == '-' })
+	if len(parts) < 3 {
+		return false
+	}
+
+	const (
+		major = iota
```

Comment:
Documentational (moved from conversation thread because github doesn't handle this properly).